crates/networking/p2p/sync.rs

```diff
@@ -114,48 +128,57 @@ impl SyncManager {
         match self.sync_mode {
             SyncMode::Snap => {
                 // snap-sync: launch tasks to fetch blocks and state in parallel
-                // - Fetch each block's state via snap p2p requests
-                // - Fetch each blocks and its receipts via eth p2p requests
-                // TODO: We are currently testing against our implementation that doesn't hold an independant snapshot and can provide all historic state
-                //       We should fetch all available state and then resort to state healing to fetch the rest
-                let (bytecode_sender, bytecode_receiver) = mpsc::channel::<Vec<H256>>(500);
-                let mut set = tokio::task::JoinSet::new();
-                set.spawn(bytecode_fetcher(
-                    bytecode_receiver,
-                    self.peers.clone(),
-                    store.clone(),
-                ));
-                set.spawn(fetch_blocks_and_receipts(
+                // - Fetch each block's body and its receipt via eth p2p requests
+                // - Fetch the pivot block's state via snap p2p requests
+                // - Execute blocks after the pivote (like in full-sync)
+                let fetch_bodies_handle = tokio::spawn(fetch_block_bodies(
                     all_block_hashes.clone(),
                     self.peers.clone(),
-                    store.clone(),
-                ));
-                let state_roots = all_block_headers
-                    .iter()
-                    .map(|header| header.state_root)
-                    .collect::<Vec<_>>();
-                set.spawn(fetch_snap_state(
-                    bytecode_sender,
-                    state_roots.clone(),
-                    self.peers.clone(),
-                    store.clone(),
                 ));
-                // Store headers
-                let mut latest_block_number = 0;
-                for (header, hash) in all_block_headers
-                    .into_iter()
-                    .zip(all_block_hashes.into_iter())
-                {
-                    latest_block_number = header.number;
-                    store.set_canonical_block(header.number, hash)?;
-                    store.add_block_header(hash, header)?;
+                let mut pivot_idx = if all_block_headers.len() > MIN_FULL_BLOCKS {
+                    all_block_headers.len() - MIN_FULL_BLOCKS
```

Comment:
Does it make sense to have all block headers in memory? For mainnet it would be ~21M. If it's a first implementation it's fine, we can optimize later

Reply:
But maybe I would just do something like `pivot_idx = head_block.number - MIN_FULL_BLOCKS`

Reply:
It made sense for small tests as we could store the full block once we had the body too, but yes, we wouldn't be able to do it with the full ethereum state